backend/chunker/chunker.go

```diff
@@ -728,6 +754,12 @@ func (f *Fs) processEntries(ctx context.Context, origEntries fs.DirEntries, dirP
 				break
 			}
 			object := f.newObject("", entry, nil)
+
+			currentXactID, err = readXactID(ctx, object)
```

Comment:
Before this change a directory listing could be finished in one network request on average on most storage providers.
This adds one extra remote file reading (and extra network roundtrip) per a composite file in a directory.
For exanple, given a subdirectory with ~100 composites and network latency ~70ms the directory listing jumps fron 0,1s to 7s.
Dilemma: we improve upload of terabate composites on S3 but slow down directory list on eg GDrive
I don't see a single solution that fits all.

Reply:
This is a good point.   It does seem that we should mitigate the problem by keeping the rename mechanism for the backends with fast rename and use this PR's approach for the other backends (Amazon S3, IBM COS, Hubic).  

The other thinking is that, when chunker is used, it is more likely that the directory will contain a relatively smaller number of large files.  So the impact of the additional object reads may be smaller.  Hopefully, some reminder/warning on the tradeoff made between slow upload and slow large-directory-listing in the S3 backend documentation can help.

In the longer term, maybe a per-directory meta object can be considered?

Reply:
In the long term chunker is only one of metadata consumers in rclone. Others are: new _compress_ layer (to seek over transparently compressed files), _crypt_ layer (to support checksums) and many backends (to simulate lacking modification time, permissions etc). Refer to https://github.com/rclone/rclone/issues?q=label%3Ametadata for use cases and https://github.com/rclone/rclone/issues?q=author%3Ayonjah+is%3Aopen for requirements. Implementing `metadir` will help only one use case but add another format to convert and another set of test cases to write and maintain. I would rather wait for an experimental Metadata interface in rclone. Probably it will be a per-remote key-value store with quasi-atomic access, locking and caching.